internal/parser/parser.go

```diff
@@ -892,6 +890,59 @@ func (p *parser) parseParenIdentExpression(lparen token.Pos, key *ast.Identifier
 	}
 }
 
+func (p *parser) parseObjectBody() *ast.ObjectExpression {
+	switch _, tok, _ := p.peek(); tok {
+	case token.IDENT:
+		ident := p.parseIdentifier()
+		return p.parseObjectBodySuffix(ident)
+	case token.STRING:
+		str := p.parseStringLiteral()
+		properties := p.parsePropertyListSuffix(str)
+		return &ast.ObjectExpression{
+			Properties: properties,
+		}
+	default:
+		// TODO(nathanielc): BadExpression.
+		return nil
+	}
+
+}
```

Comment:
For the same reason as in the grammar, in this code: if you get a string literal, it is a standard object and you can parse the property list. If it is an identifier, then you `peek` the next token, if it is a `with` than you proceed and parse the property list; if it is a `COLON` then it was a property list. Does it sound reasonable?